### DISCUSSION TITLE
How to store the control's status between different User Control?

### DISCUSSION THREAD

**Discussion by XDgl233** [AvaloniaUI/Avalonia#12391] [Q&A] [answered]
Hello, I currently run into a problem, where I set up a welcome screen in one of the sub-views in the main window(it's working through the view locator), and a button to confirm and close the welcome screen then reveals the controls behind (all packaged in a user control). But whenever I switch back from other views, the welcome screen and confirm buttons from that user control all revert back to default.

Currently, I use Community Toolkit MVVM and employ Dependency Injection. I set all the view models and views through services.AddSingleton<>(), and I presume it will make only one instance of these objects, but it doesn't seem to have any impact on the problem here. I also read some [documentation](https://github.com/AvaloniaUI/Documentation/blob/master/guides/deep-dives/reactiveui/data-persistence.md), but I am not sure I can achieve these with community toolkit MVVM.

I tried and succeed in storing the isChecked property in the view models and binding it with the button's property in two-way mode. But I assume I might encounter more similar situations in the development later: like storing the view/scroll position, remembering the checklist status, and so on. Is there a better way to achieve these? Thank you :)

**Comment by timunie** [comment]
In v11 you don't need to use the ViewLocator, but you can. However, storing the data is a task for the ViewModel. The View will reflect whatever is inside the ViewModel. If you re-create the ViewModel, it will lead to what you see. Probably you can add some logs into your VM's constructor, so you can see if a new instance was created. 

Here is a sample of Navigation: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/Routing/BasicViewLocatorSample

**Reply by XDgl233** [reply]
Yes, I follow that sample and it's very simple and works pretty smoothly in page navigation.
I know I should handle the data part myself and in ViewModels and Models. But if the information is purely View related or only about the properties of UI elements, like the view and scroll position, what should I do? Do I need to write some method to store those data myself or Avalonia or ReactiveUI have some implemented method to achieve something like page keepalive in WPF? Thank you as always, really appreciate your help, Tim :)

**Reply by timunie** [reply, marked as answer]
Probably something that you can write a Behavior for. Check out Avalonia.Behaviors.

**Reply by XDgl233** [reply]
Thank you, I will check it out.

Edit:
Update: I modified the view locator, to not create a new instance but to get service (Locator.Current.GetService<type of the view>(some parameters if needed)) by the Dependency Injection. It seems the view will be displayed and stored in memory.
